lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -518,9 +518,13 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
             final Map<String, dynamic> arguments = decoded.arguments as Map<String, dynamic>;
             // TODO(ferhat): Find more appropriate defaults? Or noop when values are null?
             final String label = arguments['label'] as String? ?? '';
-            final int primaryColor = arguments['primaryColor'] as int? ?? 0xFF000000;
             domDocument.title = label;
-            setThemeColor(ui.Color(primaryColor));
+            replyToPlatformMessage(callback, codec.encodeSuccessEnvelope(true));
```

Comment:
This may be a breaking change for many apps that expect their `setThemeColor` by this channel. I think rather than swallowing silently the `primaryColor` (if passed on the call), we should warn the user that they need to call `setSystemUIOverlayStyle` instead.

In `dom.dart`, there's a `domWindow.console.warn` that could be used for this, [like here](https://github.com/flutter/engine/blob/65515adfb5eb74f47948abf5dd9f22a13fdcf963/lib/web_ui/lib/src/engine/assets.dart#L64).

Reply:
I wrote a warning message, but the problem is that `MaterialApp` / `CupertinoApp` always provide a default `primaryColor`. For more details, see flutter/flutter#123365.

Reply:
This is great, I'm going to tweak the message a little bit so it only shows in debug (`flutter run`) mode.

Reply:
I have just wrapped the message inside `assert` to show the error message in debug mode.